ludwig/data/preprocessing.py

```diff
@@ -1205,6 +1205,50 @@ def build_dataset(
             logger.debug(f"sample {sample_ratio} of data")
             dataset_df = dataset_df.sample(frac=sample_ratio, random_state=random_seed)
 
+        # If training a reward model, perform grouping and joining on dataset
```

Comment:
Nice! This looks like the right set of transformations, but I think we'll likely want to do this at the end of preprocessing, using the processed text input feature, rather than here at the beginning. Specifically, I would consider doing this here:

https://github.com/ludwig-ai/ludwig/blob/master/ludwig/data/preprocessing.py#L1364

I would also suggest adding a test in `test_preprocessing.py` to verify it works end-to-end on a synthetic dataset.

Reply:
Sounds good! I have made the edits and refactored the code a bit to be simpler.